### ISSUE TITLE
LineEdit scaling misalignment

### ISSUE THREAD

**Issue by LoneDespair** [godotengine/godot#50105]
### Godot version

3.3.2.stable

### System information

Windows 10

### Issue description

The font and selection highlight of LineEdit becomes misaligned vertically when scaling and using 2d + expand, and is very noticeable when using small fonts

I am aware that it's not supposed to be pixel perfect but based on my test, it doesn't really affect other nodes like label and etc

Before
![Screenshot (356)](https://user-images.githubusercontent.com/35421467/124341933-ad3da900-dbf2-11eb-9f79-f47321b28a9c.png)

After
![Screenshot (357)](https://user-images.githubusercontent.com/35421467/124341935-ae6ed600-dbf2-11eb-9feb-ccf311d4a311.png)

### Steps to reproduce

1. Set stretch to 2d and expand
2. Set style to flat style
3. Test

**Comment by Calinou** [comment]
@LoneDespair Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

Also, can you reproduce this if you disable font oversampling in the Project Settings?

**Comment by LoneDespair** [comment]
> @LoneDespair Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.
> 
> Also, can you reproduce this if you disable font oversampling in the Project Settings?

- Added

**Comment by Calinou** [comment]
> Also btw, is there a way to disable oversampling on specific nodes?

No, this will be supported on specific fonts once https://github.com/godotengine/godot/pull/44772 is merged.
